src/parser/core/modules/CooldownDowntime.js

```diff
@@ -0,0 +1,183 @@
+import React from 'react'
+import {getAction} from 'data/ACTIONS'
+import Module from 'parser/core/Module'
+import {Rule, Requirement} from 'parser/core/modules/Checklist'
+import {ActionLink} from 'components/ui/DbLink'
+
+export default class CooldownDowntime extends Module {
+	static handle = 'cooldowndowntime'
+	static dependencies = [
+		'combatants',
+		'cooldowns',
+		'suggestions',
+		'checklist',
+		'downtime',
+	]
+
+	//Downtime Actions IDs
+	_downTime = {}
+	//tracking the important™ CDs
+	_trackedcds = []
+	//Default alloted time before a spell is held too long.
+	_downtimeOkTime = 0
+	//Determines if we're being overriden by an extension class
+	_extensionOverride = false
+
+	constructor(...args) {
+		super(...args)
+
+		this.addHook('begincast', {by: 'player'}, this._onBegin)
+		this.addHook('cast', {by: 'player'}, this._onCast)
+		this.addHook('complete', this._onComplete)
+	}
+
+	/**
+	 *This just sets the internal ability list if you aren't going to use override functions and
+	 perform calculated data holding in your extended class
+	 * @param {List of Abilities to track} abilities
+	 */
+	setAbilityList(abilities) {
```

Comment:
Most of the core stuff that needs extensions works by exposing  things to override, instead of call (i.e. `About`) - I'd prefer these were just "public" (no underscore prefix) class props.